### PR TITLE
Feat/gremlin

### PR DESCRIPTION
**Pull request by fantix**
- [x] Describe what this pull request does.
- [x] Make sure all tests pass.
- [x] Maintain or increase test coverage.
- [x] Black the code.
- [x] Test manually.
- [x] Remove empty sections below.
- [x] #17 merged

### New Features
- Implemented Neptune(Gremlin) export

### CODE REVIEW

File: pypfb/exporters.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                header_row = ["~label", "~id"]
+                for field in fields:
+                    field_type = "string"
+                    for field_type in field["type"]:
+                        if field_type != "null":
+                            break
+                    if isinstance(field_type, (str, unicode)):
+                        field_type = self.type_mapping[field_type]
+                    else:
+                        field_type = "String"
```
Comment (Avantol13) [inline pypfb/exporters.py]: should you log some warning/info here about defaulting to "String"?

File: pypfb/avro_utils/avro_types.py [hunk 2]
```diff
@@ -48,6 +48,17 @@ def enum(property_name, symbols, name):
     return avro_type
 
 
+def is_enum(data_type):
+    data_type = [data_type]
+    while data_type:
+        type_ = data_type.pop()
+        if isinstance(type_, list):
+            data_type.extend(type_)
+        elif isinstance(type_, dict) and type_['type'] == 'enum':
```
Comment (Avantol13) [inline pypfb/avro_utils/avro_types.py]:
```suggestion
        elif isinstance(type_, dict) and type_["type"] == "enum":
```

File: pypfb/exporters.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import os
+from uuid import uuid4
+
+from fastavro import reader
+
+from .avro_utils.avro_types import is_enum
+from .utils.str import decode
+
+
+class PFBExporter(object):
```
[the remark this answers is not in the document]
Comment (fantix) [reply pypfb/exporters.py]: oh I removed this in #20

File: pypfb/exporters.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+        if not os.path.exists(dir_path):
+            os.mkdir(dir_path)
+
+        path = os.path.join(dir_path, "gremlin_edges.csv")
+        if gzipped:
+            f = gzip.open(path + ".gz", "wb")
+        else:
+            f = open(path, "wb")
+        ew = csv.writer(f)
```
[the remark this answers is not in the document]
Comment (fantix) [reply pypfb/exporters.py]: Sure! 😄

File: pypfb/exporters.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        ew.writerow(["~id", "~from", "~to", "~label"])
+        self._handlers_by_name["~edges"] = f, ew
+
+        it = iter(self.reader)
```
[the remark this answers is not in the document]
Comment (fantix) [reply pypfb/exporters.py]: both are gone in #20 😄 `reader` in #20 is a `PFBReader`

File: pypfb/exporters.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            obj = row["object"]
+            relations = row["relations"]
+
+            pair = self._handlers_by_name.get(name)
+            if pair is None:
+                header_row = ["~label", "~id"]
+                for field in fields:
+                    field_type = "string"
+                    for field_type in field["type"]:
+                        if field_type != "null":
```
[the remark this answers is not in the document]
Comment (fantix) [reply pypfb/exporters.py]: Will do! In #20

File: pypfb/exporters.py (hunk 1 above)
Comment (fantix) [reply pypfb/exporters.py]: added a comment only, because it is for complex types like `enum`

File: pypfb/avro_utils/avro_types.py (hunk 2 above)
Comment (fantix) [reply pypfb/avro_utils/avro_types.py]: Also done in #20

**Comment by fantix** [comment]
@Avantol13 the CRF is address here: https://github.com/uc-cdis/pypfb/pull/20/commits/436d070139d305532cfc91e290743e79fa9cff64 . But if you feel we should still make improvements in this PR in case the refactor waits for too long, I'm okay too.

**Comment by fantix** [comment]
> it would be great to have some unit tests for this new code. some fake input files with a small amount of data and check the output

Oh this is also done in #20 ☀️

**Pull request merged**